services/rsources/multitenant.go

```diff
@@ -0,0 +1,50 @@
+package rsources
+
+import (
+	"context"
+	"database/sql"
+)
+
+type multitenantExtension struct {
```

Comment:
[minor] Just wondering why we called `Extension` and not `Repository`

Reply:
it is not about everything related to the database/repository, it is just about specific extension points. E.g. the select statement remains the same, it is just that it gets executed against a different database. Incrementing stats is also not part of the extension since it is common, but `createStatsTable` is, since we need to do more stuff in the multitenant scenario.